### PR TITLE
Implement missing error handling dialogs for Save to Cloud and Publish flows

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #13498

### CODE REVIEW

File: src/cloud/internal/cloudservice.cpp [hunk 1]
```diff
@@ -510,6 +515,30 @@ ProgressPtr CloudService::uploadAudio(QIODevice& audioData, const QString& audio
...
+        { 500, mu::TranslatableString("cloud", "Internal server error") },
+    };
+
+    std::string userDescription = qtrc("cloud", "Error %1: %2")
+                                  .arg(code)
+                                  .arg(mu::value(codes, code, TranslatableString("cloud", "Unknown error")).qTranslated())
+                                  .toStdString();
+
+    Ret ret = make_ret(cloud::Err::NetworkError);
+    ret.setData(CLOUD_NETWORK_ERROR_USER_DESCRIPTION_KEY, userDescription);
```
Comment (RomanPudashkin) [inline src/cloud/internal/cloudservice.cpp:539]: You forgot  `return ret;` :)

**Review by RomanPudashkin — APPROVED** [review]

**Comment by bkunda** [comment]
Tested on MacOS (both flows: "save privately" and "publish", using unactivated accounts).
All works fine! 
🟢 Ready to go from my end @RomanPudashkin 
And thanks @cbjeukendrup!

**Comment by cbjeukendrup** [comment]
@RomanPudashkin I'm planning to make one more tiny change, so maybe we should wait until that's done.

**Comment by cbjeukendrup** [comment]
Oops, too late, but no worries :)

**Comment by bkunda** [comment]
Gotta be fast Casper! Roman is hot on the merge trigger! 😁

**Comment by cbjeukendrup** [comment]
Ok, to make this story even more hilarious: turns out I confused things. I wanted to make changes to my other open "save to cloud" PR. This one was fine. Sorry for the noise 😅

**Pull request merged**